### DISCUSSION TITLE
How to turn off autoscaling?

### DISCUSSION THREAD

**Discussion by Rush-iam** [Xpra-org/xpra-html5#284] [Q&A]
I get a lot of flickering scaling in the HTML client.
When I am still - scaling is 1:1, but when I start doing anything with an app, e.g., scrolling/selecting, anything that updates the UI - I get a pixelated/heavily compressed update first, and a 1:1 normal image after some delay. It was rarely triggered for Xpra Client 8.1 (server 5.0.1), but with newer versions, it happens way more often. Especially if there is some CPU-intensive process running in the background.

I tried `--desktop-scaling=off` for the server, but it does not do anything.

**Comment by totaam** [comment]
> I tried `--desktop-scaling=off` for the server, but it does not do anything.

No, it does not. This is a client-side switch, and it doesn't do what you think it does.

> It was rarely triggered for Xpra Client 8.1

I can't think of changes that would make it use more compression than before.
Perhaps the offscreen video decoder is being used now, which means switching to video mode more readily.
Please include the server's `-d compress` log of the problematic screen updates.

**Reply by Rush-iam** [reply]
Oh, I found that setting `quality=75` and `min-speed=70` in the `/usr/share/xpra/www/default-settings.txt` file fixes the problem - now I always get non-scaled redraws and optimal performance for 4k.
I wonder how these settings relate to that scaling.
For text applications, this scaling does a bad job - the scaled text is unreadable so you need to wait for a quality redraw. I think it is better to get a non-scaled text right away, even if it has some jpeg compression artifacts - it is readable compared to the "fast" upscaled pixelated update.

**Reply by totaam** [reply]
> Oh, I found that setting quality=75

Don't do that!
At least use `min-quality` - NOT `quality`.

**Reply by totaam** [reply]
Text applications should not be scaled, ever. That would be a bug.

**Reply by totaam** [reply]
The non-lossless text applications bug should have been fixed in v5.0.5: https://github.com/Xpra-org/xpra/commit/1aaf6ca3a95390cf3a521b08b2298ded8f96026d
But the commit was incorrect, the 5.0.5-r1 builds have the fix: https://github.com/Xpra-org/xpra/commit/1aaf6ca3a95390cf3a521b08b2298ded8f96026d
